pyHS100/discover.py

```diff
@@ -53,14 +54,19 @@ def discover(protocol: TPLinkSmartHomeProtocol = None,
                 ip, port = addr
                 info = json.loads(protocol.decrypt(data))
                 device_class = Discover._get_device_class(info)
-                if device_class is not None:
+                if deviceType is not None:
+                    if deviceType in str(device_class):
+                        if device_class is not None:
+                            devices[ip] = device_class(ip)
+                elif device_class is not None:
                     devices[ip] = device_class(ip)
+                else:
+                    return None
```

Comment:
This line looks like a breaking change. Before we did not return `None` but the `devices` variable in https://github.com/GadgetReactor/pyHS100/blob/07b4eda7ae3250c3df826de6f1dbd6b6c5f873ef/pyHS100/discover.py#L62

In turn devices is initialized as an empty dict https://github.com/GadgetReactor/pyHS100/blob/07b4eda7ae3250c3df826de6f1dbd6b6c5f873ef/pyHS100/discover.py#L47

Reply:
It's not really a breaking change since it only returns None if you specify the deviceType. The original return is still there. However, I will change it to be consistent.